pkg/download/classic/download_test.go

```diff
@@ -305,57 +184,127 @@ func TestDownload_FilteringCanBeTurnedOffViaFeatureFlags(t *testing.T) {
 		},
 	}}
 
-	testAPI1 := api.API{ID: "API_ID_1", URLPath: "API_PATH_1", NonUniqueName: true}
-	testAPI2 := api.API{ID: "API_ID_2", URLPath: "API_PATH_2", NonUniqueName: false}
-	apiMap := api.APIs{"API_ID_1": testAPI1, "API_ID_2": testAPI2}
-
-	configurations, err := Download(c, "project", apiMap, filters)
+	configurations, err := Download(c, "project", toAPIs(api1, api2), filters)
 	assert.NoError(t, err)
 	assert.Len(t, configurations, 1)
 }
 
-func TestDownload_APIWithoutAnyConfigAvailableAreNotDownloaded(t *testing.T) {
-	c := dtclient.NewMockClient(gomock.NewController(t))
-	c.EXPECT().ListConfigs(gomock.Any(), gomock.Any()).DoAndReturn(func(ctx context.Context, a api.API) ([]dtclient.Value, error) {
-		if a.ID == "API_ID_1" {
-			return []dtclient.Value{{Id: "API_ID_1", Name: "API_NAME_1"}}, nil
-		} else if a.ID == "API_ID_2" {
-			return []dtclient.Value{}, nil
-		}
-		return nil, nil
-	}).Times(2)
-	c.EXPECT().ReadConfigById(gomock.Any(), gomock.Any()).Return([]byte("{}"), nil)
+func Test_generalCases(t *testing.T) {
```

Comment:
nit: I'm not convinced of the benefit these being a table test here

Reply:
I understand your problems, but let's go for now with this...